### ISSUE TITLE
Need to create a find function in DB Models

### ISSUE THREAD

**Issue by martinpetrov1096** [ai-api/backend#8]
When creating a user, I need to be able to check if a username is already taken so I can notify the user. 
Also, when the user enters their credentials, I need to find the user by username. So we definitely need functionality to search by username, but it might be a good time to think of implementing a generic find function in the db models. Though I can also see this being a lot of work, in which case we can just add search functions for specific fields as we need them

**Comment by rdey0** [comment]
So right now I'm thinking there are three ways to go about doing this
1. Create a generic find function in the data model objects which will find a data model based on given parameters
2. Create a function in `db/dbOperations` which checks if a given username already exists
3. Have the User data model class check if the username already exists before creating a new entry

Because the service objects are the ones which do all of the data validation, I think option 3 is not the responsibility of the data model objects and option 2 seems too specific of a task to be included in dbOperations. So I think going with option 1 and creating a general find function is the better choice.

Right now I'm thinking we can do this by creating a static function in all data model objects which will look something like this:
``` javascript
public static async find(client: PoolClient, columnNames: Array<string>, columnValues: Array<unknown>): Promise<Array<Record<string, unkown>>>{
     // returns an array containing table entries matching the given parameters in the form of objects
}
```
So for example, if you want to find specific user entries you would call:
``` javascript
User.find(client, ['columnName'], ['columnValue'])
```
This would return an array of objects. Each object would be a table entry which matches your query. You can then use this static function to create useful helper functions in your Service objects, like checking if a username already exists. Let me know what you think about this.

**Comment by martinpetrov1096** [comment]
@rdey0 I agree entirely that option 1 seems to be the best one. How long do you think it'll take to implement?

**Comment by rdey0** [comment]
I should be able to finish it within a day or two. I'm planning on finishing up adding the PackageFlag data model and then I'll add the general find feature. I'll make a pull request when I'm done.
